controlplane/kubeadm/controllers/alias.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"time"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	"sigs.k8s.io/cluster-api/controllers/remote"
+	kubeadmcontrolplanecontrollers "sigs.k8s.io/cluster-api/controlplane/kubeadm/internal/controllers"
+)
+
+// KubeadmControlPlaneReconciler reconciles a KubeadmControlPlane object.
+type KubeadmControlPlaneReconciler struct {
+	Client    client.Client
+	APIReader client.Reader
+	Tracker   *remote.ClusterCacheTracker
+
+	EtcdDialTimeout time.Duration
+
+	// WatchFilterValue is the label value used to filter events prior to reconciliation.
+	WatchFilterValue string
+}
+
+// SetupWithManager sets up the reconciler with the Manager.
+func (r *KubeadmControlPlaneReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
+	return (&kubeadmcontrolplanecontrollers.KubeadmControlPlaneReconciler{
```

Comment:
to be consistent with the other packages we moved, do we want to take the chance to rename this to `kubeadmcontrolplanecontrollers.Reconciler`?

Reply:
It looks to me like we only did this for the core-controller where the package name is the resource name. 

The controllers moved into internal/controllers still have the full name because of `controllers.`, e.g. `KubeadmConfigReconciler`, `DockerClusterReconciler`.

I think the idea is:
* if we only have a few controllers and it's not too complicated, keep the controllers in internal/controllers and still call them `controllers.ResourceReconciler`.
* if it get's to crazy create separate sub-packages per resource and then rename the struct, e.g. cluster.Reconciler

Reply:
/cc @fabriziopandini

Reply:
agreed, this is consistent with CABPK, CAPD, experiments